AutonomousDemo/VehicleMotion.cs

```diff
@@ -19,7 +19,7 @@ public VehicleMotion(double currentSpeed)
             this.CurrentSpeed = currentSpeed;
         }
 
-        public Velocity Drive(double meters, int seconds)
+        public Velocity CalculateCurrentVelocity(double meters, int seconds)
```

Comment:
So... this function is still pretty jacked. Consider:

* You pass in meters and seconds. From that, it determines the desired final velocity.
* Acceleration is determined by assuming you will take *that same number of seconds* to reach the final velocity. So we're using "seconds" to mean two things here.
* We then apply that acceleration, again for that same number of seconds... so we therefore return the velocity of meters / seconds. I.e., this could be replaced without change by:

```
public Velocity CalculateCurrentVelocity(double meters, int seconds)
{
    return new Velocity(meters / seconds);
}
```

Also, running this function has the side-effect of setting CurrentAcceleration, a property on this class, but you declare a *local variable* of the same name as another class-level property, CurrentVelocity, which I'm a little surprised isn't throwing red flags in your IDE, but admittedly I do not have a working copy of Visual Studio at the moment.

I would go back to the drawing board and think about what you're trying to accomplish. Maybe this is just trying to do too many things; perhaps a better function would take in a desired velocity and the time to reach it, and use it to assign acceleration, and then a *separate* function would apply acceleration over some unit time.